### PR TITLE
Added the test for GetProject

### PR DESCRIPTION
**Pull request by Justinfan591**
Please make some comments.

### CODE REVIEW

**Comment by Justinfan591** [comment]
I left Dennis out since he has COVID.

**Comment by Raqkkkkk** [comment]
> I left Dennis out since he has COVID.

poor Dennis ;-;

**Review by Raqkkkkk — APPROVED** [review]
lgtm 😄

**Review by cecexyzhang — APPROVED** [review]
The tests have improved our code coverage. Looks good.

File: src/test/java/usecase/getprojects/GetProjectsInteractorTest.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        HashSet<Integer> projectIds = new HashSet<>();
+        projectIds.add(101);
+
+        when(mockUserProjectsRepository.getProjectIdsForUser(userId)).thenReturn(projectIds);
+        when(mockProjectRepository.getProjectById(101)).thenReturn(null);
+
+        GetProjectsInputData inputData = new GetProjectsInputData(userId);
+        interactor.getProjects(inputData);
+
+        verify(mockGetProjectsPresenter).prepareFailView(anyString());
```
Comment (Raqkkkkk) [inline src/test/java/usecase/getprojects/GetProjectsInteractorTest.java]: After some intense testing and debugging (pulling to my computer, running it, and looking at it), this test will fail.
I can currently find two causes:
- The interactor is not using the mock repository (will be addressed in #124), will need slight modification to setup after it gets merged
- If the projectRepository returns null, it will create a null pointer exception based on current implementation of the interactor. We should modify the tests and/or the implementation so that it matches what we expect in this scenario.

**Review by Raqkkkkk — CHANGES_REQUESTED** [review]
Noticed a failed test

File: src/test/java/usecase/getprojects/GetProjectsInteractorTest.java (hunk 1 above)
Comment (Justinfan591) [reply src/test/java/usecase/getprojects/GetProjectsInteractorTest.java:85]: I'll try to figure it out, but I have little clue with the projectRepo part. If doesn't work then someone please help.

**Review by Pixeldom4 — APPROVED** [review]
Generally I appreciate the good organization and consistent structure of your code, each test class includes setup methods initialized with @BeforeEach. Good usage of Mockito and JUnit5 frameworks! 

***Some Suggestions***:
**GetProjectsControllerTest**: I think that these test cases should test for varying user ID edge cases 

**GetProjectsPresenterTest**: I think that you should test null and empty data sets as potential edge cases

Note: These suggestions are entirely optional and not necessary, but they would help improve the holistic quality of the tests

**Comment by Raqkkkkk** [comment]
will merge main to this pull request to make sure it works with modification of controllers in #114

File: src/test/java/usecase/getprojects/GetProjectsInteractorTest.java (hunk 1 above)
Comment (Raqkkkkk) [reply src/test/java/usecase/getprojects/GetProjectsInteractorTest.java]: resolved by letting the interactor ignore null projects

**Review by Raqkkkkk — APPROVED** [review]

**Comment by Raqkkkkk** [comment]
> Generally I appreciate the good organization and consistent structure of your code, each test class includes setup methods initialized with @beforeeach. Good usage of Mockito and JUnit5 frameworks!
> 
> _**Some Suggestions**_: **GetProjectsControllerTest**: I think that these test cases should test for varying user ID edge cases
> 
> **GetProjectsPresenterTest**: I think that you should test null and empty data sets as potential edge cases
> 
> Note: These suggestions are entirely optional and not necessary, but they would help improve the holistic quality of the tests

I think this should be fine since neither the controller or presenter deals with the data (the controller just package it in inputData and sends in to the interactor while the presenter unpackages outputData and sends it to the view model) hence as long as the parameters matches what is expected everything should be the same, no matter if the data being unpackaged contains nothing or is null (should let view model deal with it).
